### PR TITLE
docs: update datasets examples

### PR DESCRIPTION
**Pull request by jmtaylor90**
Make sure these boxes are signed before submitting your Pull Request -- thank you.

- [x] I have read the contributing guide lines at https://redmine.openinfosecfoundation.org/projects/suricata/wiki/Contributing
- [x] I have signed the Open Information Security Foundation contribution agreement at https://suricata-ids.org/about/contribution-agreement/
- [x] I have updated the user guide (in doc/userguide/) to reflect the changes made (if applicable)

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket:

Describe changes:
- updates datasets examples
- updates datarep syntax to match rule
-

[PRScript](https://redmine.openinfosecfoundation.org/projects/suricata/wiki/PRscript) output (if applicable):

### CODE REVIEW

File: doc/userguide/rules/datasets.rst [hunk 1]
```diff
@@ -1,9 +1,6 @@
 Datasets
 ========
 
-.. warning:: This is an experimental feature. Syntax and functionality may
```
Comment (victorjulien) [inline doc/userguide/rules/datasets.rst]: this should stay

File: doc/userguide/rules/datasets.rst [hunk 2]
```diff
@@ -34,8 +31,8 @@ Example of sets for tracking unique values::
 
 Rules to go with the above::
 
-    http.user_agent; dataset:set,ua-seen;
-    dns.query; to_sha256; dataset:set,dns-sha256-seen;
+    alert http any any -> any any (msg: "http user-agent test"; http.user_agent; dataset:set,ua-seen; sid:234; rev:1;)
+    alert dns any any -> any any (msg:"dns list test"; dns.query; to_sha256; dataset:isset,dns-sha256-seen; sid:123; rev:1;)
```
Comment (victorjulien) [inline doc/userguide/rules/datasets.rst]: Better to use
```
.. container:: example-rule
```

Comment (victorjulien) [reply doc/userguide/rules/datasets.rst]: See rules/header-keywords.rst for some examples of how it is used.

Comment (jmtaylor90) [reply doc/userguide/rules/datasets.rst]: updated thanks!

File: doc/userguide/rules/datasets.rst (hunk 1 above)
Comment (jmtaylor90) [reply doc/userguide/rules/datasets.rst]: oops, added back

**Comment by jmtaylor90** [comment]
continued in #4317

**Pull request closed without merging**